<launch>

<arg name = 'cmd_src' default = 'teleop' />
<arg name = 'robot' default = 'nusim' />
<arg name = 'rviz' default = 'true' />
<arg name = 'nuslam' default= 'true'/>

<param name= "rate" type="double" value="100"/>
<param name= "data_association" type="int" value="0"/>
<param name="wheel_left" type="string" value="red_wheel_left_joint" />
<param name="wheel_right" type="string" value="red_wheel_right_joint" />
<param name="body_id" type="string" value="red_base_footprint" />
<param name="odom_id" type="string" value="odom" />
<rosparam command="load" file="$(find nusim)/config/initial.yaml" />
<rosparam command="load" file="$(find nusim)/config/sensor.yaml" />
<rosparam command="load" file="$(find nuturtle_description)/config/diff_params.yaml" />

<group if="$(eval robot == 'nusim')">
    <include file="$(find nuturtle_description)/launch/load.launch">
        <arg name="color" value="red"/>
        <arg name = 'use_rviz' value = 'false' />
        <arg name = 'use_jsp' value = 'false' />
    </include>

    <node name="turtle_interface" pkg="nuturtle_control" type="turtle_interface" output = "screen"/> 

    <node name="nusim_node" pkg="nusim" type="nusim_node" output = "screen"> 
        <rosparam command="load" file="$(find nusim)/config/initial.yaml" />
    </node>
</group>

<group if="$(eval robot == 'localhost')">
    <include file="$(find nuturtle_control)/launch/basic_remote.launch"/>

    <node name="turtle_interface" pkg="nuturtle_control" type="turtle_interface" output = "screen"/> 
</group>

<group if="$(arg nuslam)">
    <node name="nuslam_node" pkg="nuslam" type="nuslam_node" output = "screen"/> 
</group>

<include file="$(find nuturtle_description)/launch/load.launch">
    <arg name="color" value="blue"/>
    <arg name = 'use_rviz' value = 'false' />
    <arg name = 'use_jsp' value = 'true' />
</include>

<include file="$(find nuturtle_description)/launch/load.launch">
    <arg name="color" value="green"/>
    <arg name = 'use_rviz' value = 'false' />
    <arg name = 'use_jsp' value = 'true' />
</include>

<group if="$(eval cmd_src == 'circle')">
    <node name="circle" pkg="nuturtle_control" type="circle" output = "screen" /> 
</group>

<group if="$(eval cmd_src == 'teleop')">
    <node pkg="turtlebot3_teleop" type="turtlebot3_teleop_key" name="turtlebot3_teleop_keyboard"  output="screen"/>
</group>

<node pkg="tf2_ros" type="static_transform_publisher" name="world_to_odom" args="0 0 0 0 0 0 world odom" />


<group if="$(arg rviz)">
    <node name="rviz" pkg ="rviz" type="rviz" args="-d $(find nusim)/config/nusim.rviz" />
    <!-- <node name="rviz" pkg ="rviz" type="rviz" /> -->
</group>

<group if="$(eval robot == 'raphael')">
    <node name="turtle_interface" pkg="nuturtle_control" type="turtle_interface" output = "screen" machine="turtlebot"/> 

    <include file="$(find nuturtle_control)/launch/basic_remote.launch">
        <arg name = "robot" value = "raphael" />
    </include>
</group>

</launch>